### ISSUE TITLE
Current DataGrid is not accessible by a screen reader.

### ISSUE THREAD

**Issue by mvaden** [mui/mui-x#9075]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Currently, `<DataGrid />` (DataGrid) does not provide property for handling various ARIA attributes to made this component more accessibly for screen reader technologies. Even when I attempt to wrap the component in an element with `aria-live="polite"`, the component is not detected by a screen reader, which is key for Section 508 compliance. It would be great to have a DataGrid component who's columns, rows, table headers, table footers and table cells and all be read by a screen reader and also be tabbed by default or be set by a propriety on the component itself. Currently, neither is possible; all that is possible is setting `aria-label` or `aria-labelledby`.

At the time that this ticket is produced, we are use `"@mui/x-data-grid": "^6.0.3",`.

### Examples 🌈

The feature that is being requested here does not relate to any CSS style or design changes or implemntations.

### Motivation 🔦

Currently, we are trying to leverage the use of MUI and it's features in a product that must conform to Section 508 standards, since this will eventually be owned and maintained by the government.
